### PR TITLE
Door colliders

### PR DESCRIPTION
**Pull request by jfc978**
Door Colliders and room changing now work. There are issues with the doors when walking through walls, which is more of an issue for the ability to walk through walls.

### CODE REVIEW

File: CrossPlatformDesktopProject/CrossPlatformDesktopProject/Levels/Background.cs [hunk 1]
```diff
@@ -103,5 +104,26 @@ public void DoorCheck(SpriteBatch sb, string[] adjacentRooms, Texture2D texture)
...
+            if (!adjacentRooms[2].Equals("-1"))
+            {
+                doors.Add(new Door(new Rectangle(leftDoorDest.Center.X, 0, 1, screenHeight), currentMap, adjacentRooms[2]));
+            }
+            if (!adjacentRooms[3].Equals("-1"))
+            {
+                doors.Add(new Door(new Rectangle(rightDoorDest.Center.X, 0, 1, screenHeight), currentMap, adjacentRooms[3]));
+            }
+            return doors;
+        }
```
Comment (rustbeltdurt) [inline CrossPlatformDesktopProject/CrossPlatformDesktopProject/Levels/Background.cs:127]: This & DoorCheck() could probably be cleaned up together, but we can worry about that during refactoring after the sprint.

**Comment by rustbeltdurt** [comment]
All the code looks good from what I can see & it works (as I expected) on my machine. Can't merge yet because of conflicts in Game1.cs. Aside from my one cleaning comment in Background.cs this all seems good.

**Comment by sweeneyde** [comment]
I think doing my sword PR first might help clear up the conflicts.

**Comment by sweeneyde** [comment]
Oops I was wrong it made it worse.

I resolve the conflicts.

**Comment by sweeneyde** [comment]
How's that?

**Comment by sweeneyde** [comment]
Ope still some compile errors to fix

**Comment by bhasseman** [comment]
I got errors from the Map class because of accessibility errors (some things are public while others are not)

**Comment by sweeneyde** [comment]
This still has the error 'Constructor on type 'CrossPlatformDesktopProject.NPC.Goriya' not found.' whenever mouse-clicking

**Comment by bhasseman** [comment]
It gets the same error when walking through the door - unless we change every room csv to have a bat instead of Goriya, we won't be able to test it until Roy finishes the Goriya implementations

**Pull request merged**